### PR TITLE
Fix some locking errors due to the unreleased locks

### PR DESCRIPTION
**Pull request by anson-lo**
Fix some (small) locking errors due to the unreleased locks before program exit for better resource management and good code symmetry reported in #1600 #1601 #1602.

### CODE REVIEW

**Comment by adrianreber** [comment]
Thanks for the fixes. Please have a look at the errors reported by our CI. Also, please look at https://github.com/checkpoint-restore/criu/blob/criu-dev/CONTRIBUTING.md for correctly formatting your commits.

File: test/zdtm/static/seccomp_filter_tsync.c [hunk 1]
```diff
@@ -158,6 +160,7 @@ int main(int argc, char **argv)
 		 * inherited.
 		 */
 		if (filter_syscall(__NR_ptrace, SECCOMP_FILTER_FLAG_TSYNC) < 0)
+			pthread_mutex_unlock(&getpid_wait);
```
Comment (avagin) [inline test/zdtm/static/seccomp_filter_tsync.c:163]: Have you tested these changes? _exit is called unconditionally...

File: test/zdtm/static/seccomp_filter_tsync.c [hunk 2]
```diff
@@ -145,11 +145,13 @@ int main(int argc, char **argv)
 
 		if (write(sk, &c, 1) != 1) {
 			pr_perror("write");
+			pthread_mutex_unlock(&getpid_wait);
```
[the remark this answers is not in the document]
Comment (anson-lo) [reply test/zdtm/static/seccomp_filter_tsync.c:148]: There is also no harm to add unlock statement before exit(1) to release the lock before for better resource management and for better codes. Also, taking locks before program exit might slow down termination.

File: test/zdtm/static/seccomp_filter_tsync.c (hunk 1 above)
Comment (anson-lo) [reply test/zdtm/static/seccomp_filter_tsync.c:163]: I don't know why but I am trying my best to resolve these

**Comment by anson-lo** [comment]
Let me create a new PR  and resolve all the errors reported by our CI soon ...

**Pull request closed without merging**